crates/ruff_linter/src/rules/ruff/rules/spurious_async.rs

```diff
@@ -0,0 +1,83 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::visitor;
+use ruff_python_ast::visitor::Visitor;
+use ruff_python_ast::{self as ast, Expr, Stmt};
+use ruff_text_size::TextRange;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for functions declared `async` that do not yield or otherwise use features requiring the
+/// function to be declared `async`.
+///
+/// ## Why is this bad?
+/// Declaring a function `async` when it's not is usually a mistake, and may artificially limit the
+/// contexts where that function may be called.
```

Comment:
I think we'll want to include an example here too

Reply:
You can also probably say "will limit the contexts" instead of "may artificially limit".

There's also something to be said about it actually being _bad_ to say that your function is async without it containing a yield as it's breaking the fundamental promise of the async label. Some more context on this idea [in the trio documentation](https://trio.readthedocs.io/en/stable/reference-core.html#checkpoints). I'm not sure if we can get into that without confusing users though.

Reply:
I've made this change below.